ssl/base/src/main/java/org/wildfly/security/ssl/DomainlessSSLContextBuilder.java

```diff
@@ -0,0 +1,337 @@
+/*
+ * Copyright 2022 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.ssl;
+
+import org.wildfly.common.Assert;
+import org.wildfly.security.FixedSecurityFactory;
+import org.wildfly.security.OneTimeSecurityFactory;
+import org.wildfly.security.SecurityFactory;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSessionContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509ExtendedKeyManager;
+import javax.net.ssl.X509KeyManager;
+import javax.net.ssl.X509TrustManager;
+import java.security.Provider;
+import java.security.Security;
+import java.util.function.Supplier;
+
+import static org.wildfly.security.provider.util.ProviderUtil.INSTALLED_PROVIDERS;
+
+/**
+ * A class which allows building and configuration of a single client- or server-side SSL context.  The builder requires, at a
+ * minimum, that a key manager be set; all other parameters have default values as follows:
+ * <ul>
+ *     <li>The cipher suite selector defaults to {@link CipherSuiteSelector#openSslDefault()}</li>
+ *     <li>The protocol suite selector defaults to {@link ProtocolSelector#DEFAULT_SELECTOR}</li>
+ *     <li>The "require client authentication" flag defaults to {@code false}</li>
+ *     <li>The provider supplier defaults to {@link Security#getProviders() Security::getProviders}</li>
+ * </ul>
+ *
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+public final class DomainlessSSLContextBuilder {
```

Comment:
Just trying to think of some possible names here to make it more clear what this class is about. Perhaps `BaseSSLContextBuilder`?

Reply:
yeah, I like that name better. I'll also update both Javadocs to explain the difference.

Reply:
Renamed to `BaseSSLContextBuilder`, and also reviewed the files for new changes. There were a couple in the test cases, but since they are identical between the modules, I just created a test-jar for `wildfly-elytron-ssl-base` and called the code directly (similar to what I did in wildfly/wildfly-core#5561).